src/Storages/StorageMongoDB.cpp

```diff
@@ -28,6 +35,49 @@ namespace ErrorCodes
 {
     extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
     extern const int MONGODB_CANNOT_AUTHENTICATE;
+    extern const int MONGODB_ERROR;
+    extern const int EMPTY_DATA_PASSED;
+}
+
+namespace
+{
+    using MongoArray = Poco::MongoDB::Array;
+
+    DataTypePtr getDataTypeMongoDB(const Poco::MongoDB::Element & value)
+    {
+        switch (value.type())
+        {
+            case Poco::MongoDB::ElementTraits<Poco::MongoDB::NullValue>::TypeId:
+                return std::make_shared<DataTypeNothing>();
+            case Poco::MongoDB::ElementTraits<bool>::TypeId:
+                return std::make_shared<DataTypeUInt8>();
+            case Poco::MongoDB::ElementTraits<Int32>::TypeId:
+                return std::make_shared<DataTypeInt32>();
+            case Poco::MongoDB::ElementTraits<Poco::Int64>::TypeId:
+                return std::make_shared<DataTypeInt64>();
+            case Poco::MongoDB::ElementTraits<Float64>::TypeId:
+                return std::make_shared<DataTypeFloat64>();
+            case Poco::MongoDB::ElementTraits<Poco::Timestamp>::TypeId:
+                return std::make_shared<DataTypeDateTime>();
+            case Poco::MongoDB::ElementTraits<MongoArray::Ptr>::TypeId:
+            {
+                auto parent = static_cast<const Poco::MongoDB::ConcreteElement<MongoArray::Ptr> &>(value).value();
+
+                DataTypes types;
+                types.reserve(parent->size());
+
+                for (size_t idx = 0, size = parent->size(); idx < size; ++idx)
+                {
+                    Poco::MongoDB::Element::Ptr child = parent->get(static_cast<int>(idx));
+                    types.push_back(getDataTypeMongoDB(*child));
+                }
+                return std::make_shared<DataTypeTuple>(types);
```

Comment:
As I can see we don't support reading Array with different types as Tuple from Mongo. It's a good idea to implement it later, but now we should check that all nested types are the same or have common type (using `getLeastSuperType`) and throw an exception if there is no common type or create an Array(comon_type).

Reply:
Maybe I miss somethingm but I still don't see support for reading Array with different types as Tuple from Mongo